src/generic_containers.rs

```diff
@@ -0,0 +1,114 @@
+//! **Private** generalizations of containers:
+//! - `Map`: `BTreeMap` and `HashMap` (any hasher).
+
+#![cfg(feature = "use_alloc")]
+
+use alloc::collections::BTreeMap;
+use alloc::vec::Vec;
+#[cfg(feature = "use_std")]
+use core::hash::{BuildHasher, Hash};
+#[cfg(feature = "use_std")]
+use std::collections::HashMap;
+
+pub trait Map {
+    type Key;
+    type Value;
+    fn is_empty(&self) -> bool;
+    fn clear(&mut self);
```

Comment:
I would remove these since they are pretty much unused.

Reply:
I planned on removing the unused one, but will remove both indeed.